src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = _curry1(function uniq(list) {
+  /* global Set */
+  var item, set, idx, len, items, uniqs;
+
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  if (typeof Set === 'function') {
```

Comment:
I originally tried that with something like the following, though there is something whacky with the build tool, where it doesn't include the separate `uniq` function in `dist/ramda.js` (and also pulls all inline comments out above).
```js
function uniq(list) {/*...*/}
module.exports = typeof Set === 'function' ? _curry1(uniq) : uniqWith(equals);
```
results in `dist/ramda.js` only containing:
```js
    /* global Set */
    /**
       * Uses a native `Set` instance where possible for
       * removing duplicate items. Items that implement
       * the fantasy-land Setoid spec fallback to using
       * `_contains` to support custom equality behaviour.
       */
    // -0 must be handled separately, as Set treats 0 and -0 as equal.
    /**
     * Returns a new list containing only one copy of each element in the original list.
     * `R.equals` is used to determine equality.
     *
     * @func
     * @memberOf R
     * @category List
     * @sig [a] -> [a]
     * @param {Array} list The array to consider.
     * @return {Array} The list of unique items.
     * @example
     *
     *      R.uniq([1, 1, 2, 1]); //=> [1, 2]
     *      R.uniq([1, '1']);     //=> [1, '1']
     *      R.uniq([[42], [42]]); //=> [[42]]
     */
    var uniq = typeof Set === 'function' ? _curry1(uniq) : uniqWith(equals);
```

Is there a "correct" way for handling this?